otherlibs/dune-rpc/private/dune_rpc_private.mli

```diff
@@ -121,32 +121,64 @@ module Loc : sig
     }
 end
 
-module Error : sig
+module Diagnostic : sig
+  type severity =
+    | Error
+    | Warning
+
+  module Promotion : sig
+    type t =
+      { in_build : string
+      ; in_source : string
+      }
+  end
+
   type t =
     { target : string option
     ; message : string
     ; loc : Loc.t option
+    ; severity : severity option
+    ; promotion : Promotion.t option
+          (* TODO should this be generalied to a list? *)
```

Comment:
What is the alternative? Do you send one diagnostic per promotion?

Reply:
Some users asked to have more than 1 promotion per failed rule. E.g. `(progn (diff ..) (diff ..))`, but without failing on the first `diff`.